pkg/plugins/gateway/algorithms/pd_disaggregation.go

```diff
@@ -180,11 +189,32 @@ func (r *pdRouter) filterPrefillDecodePods(routingCtx *types.RoutingContext, rea
 		case "decode":
 			decodePods = append(decodePods, pod)
 		}
+
+		if aibrixPromptLengthBucketing {
```

Comment:
Seems the codes from 197-201 is duplicated. 

is the code a filter? If so, can we write in similar way like this?

```
		// For multi-node scenarios, only select pods from node_rank=0 (PodGroupIndex=0)
		// which have the HTTP server running
		if !isPodWithHTTPServer(pod) {
			continue
		}
```

```
if aibrixPromptLengthBucketing {
    if !r.isPodSuitableForPromptLength(pod, promptLength) {
				continue
			}
}
```